# Introduction To The Dataset

In [1]:
import pandas as pd

data=pd.read_csv("thanksgiving.csv",encoding="Latin-1")

In [2]:
data.head()

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                                                NaN                                      
3                                                NaN                                      
4                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   
1                                  Baked   
2                                Roasted   
3                                  Baked   
4                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                
1                                                NaN                
2                                                NaN                
3                                                NaN                
4                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      
1                                        Bread-based      
2                                         Rice-based      
3                                        Bread-based      
4                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               
1                                                NaN                               
2                                                NaN                               
3                                                NaN                               
4                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   
1                             Other (please specify)   
2                                           Homemade   
3                                           Homemade   
4                                             Canned   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            
1                    Homemade cranberry gelatin ring                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

          ...          \
0         ...           
1         ...           
2         ...           
3         ...           
4         ...           

  Have you ever tried to meet up with hometown friends on Thanksgiving night?  \
0                                                Yes                            
1                                                 No                            


In [3]:
print(data.columns)


Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

# Filtering Out Rows From A DataFrame

In [4]:
data["Do you celebrate Thanksgiving?"].value_counts(dropna = False)

Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64

In [5]:
data=data[data["Do you celebrate Thanksgiving?"]=="Yes"]

#  Using Value_counts To Explore Main Dishes

In [6]:
data["What is typically the main dish at your Thanksgiving dinner?"].value_counts(dropna = False)

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
NaN                         6
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

In [7]:
data[data["What is typically the main dish at your Thanksgiving dinner?"] == "Tofurkey"]["Do you typically have gravy?"]

4      Yes
33     Yes
69      No
72      No
77     Yes
145    Yes
175    Yes
218     No
243    Yes
275     No
393    Yes
399    Yes
571    Yes
594    Yes
628     No
774     No
820     No
837    Yes
860     No
953    Yes
Name: Do you typically have gravy?, dtype: object

# Figuring Out What Pies People Eat

In [8]:
apple = "Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple"
pumpkin = "Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin"
pecan = "Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan"

(data[apple].notnull() | data[pumpkin].notnull() | data[pecan].notnull()).value_counts()

True     876
False    104
dtype: int64

# Findings
1. 876 people eat atleast one of the three pies
2. 104 people eat none of the pies

# Converting Age To Numeric

In [9]:
data["Age"].value_counts(dropna = False)

45 - 59    269
60+        258
30 - 44    235
18 - 29    185
NaN         33
Name: Age, dtype: int64

In [10]:
def extract_age(age_str):
    if pd.isnull(age_str):
        return None
    age_str = age_str.split(" ")[0]
    age_str = age_str.replace("+", "")
    return int(age_str)

data["int_age"] = data["Age"].apply(extract_age)

In [11]:
data["int_age"].describe()

count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%       30.000000
50%       45.000000
75%       60.000000
max       60.000000
Name: int_age, dtype: float64

# Findings

Although we only have a rough approximation of age, and it skews downward because we took the first value in each string (the lower bound), we can see that that age groups of respondents are fairly evenly distributed.

# Converting Income To Numeric

In [12]:
data["How much total combined money did all members of your HOUSEHOLD earn last year?"].value_counts(dropna = False)

$25,000 to $49,999      166
$75,000 to $99,999      127
$50,000 to $74,999      127
Prefer not to answer    118
$100,000 to $124,999    109
$200,000 and up          76
$10,000 to $24,999       60
$0 to $9,999             52
$125,000 to $149,999     48
$150,000 to $174,999     38
NaN                      33
$175,000 to $199,999     26
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: int64

In [13]:
def extract_income(income_str):
    if pd.isnull(income_str) or "Prefer" in income_str :
        return None
    income_str=income_str.split(" ")[0]
    income_str=income_str.replace("$","")
    income_str=income_str.replace(",","")
    return int(income_str)
    
data["int_income"]=data["How much total combined money did all members of your HOUSEHOLD earn last year?"].apply(extract_income)

In [14]:
data["int_income"].describe()

count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%       25000.000000
50%       75000.000000
75%      100000.000000
max      200000.000000
Name: int_income, dtype: float64

# Findings
Although we only have a rough approximation of income, and it skews downward because we took the first value in each string (the lower bound), the average income seems to be fairly high, although there is also a large standard deviation.

# Correlating Travel Distance And Income

In [15]:
data[data["int_income"]<50000]["How far will you travel for Thanksgiving?"].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         106
Thanksgiving is local--it will take place in the town I live in                      92
Thanksgiving is out of town but not too far--it's a drive of a few hours or less     64
Thanksgiving is out of town and far away--I have to drive several hours or fly       16
Name: How far will you travel for Thanksgiving?, dtype: int64

In [16]:
data[data["int_income"]>150000]["How far will you travel for Thanksgiving?"].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         49
Thanksgiving is local--it will take place in the town I live in                     25
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    16
Thanksgiving is out of town and far away--I have to drive several hours or fly      12
Name: How far will you travel for Thanksgiving?, dtype: int64

# Findings
It appears that more people with high income have Thanksgiving at home than people with low income. This may be because younger students, who don't have a high income, tend to go home, whereas parents, who have higher incomes, don't.

# Linking Friendship And Age

In [17]:
data.pivot_table(index="Have you ever tried to meet up with hometown friends on Thanksgiving night?"
                 ,columns='Have you ever attended a "Friendsgiving?"',values="int_age")

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [18]:
data.pivot_table(index="Have you ever tried to meet up with hometown friends on Thanksgiving night?"
                 ,columns='Have you ever attended a "Friendsgiving?"',values="int_income")

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842

# Calculating Number Of People Working on Thanksgiving

In [19]:
data['Will you employer make you work on Black Friday?'].value_counts(dropna = False)

NaN              910
Yes               43
No                20
Doesn't apply      7
Name: Will you employer make you work on Black Friday?, dtype: int64

# Finding Regional Patterns in Dinner Menus

In [20]:
data.pivot_table(index = ['What is typically the main dish at your Thanksgiving dinner?','US Region'])

RespondentID  \
What is typically the main dish at your Thanksg... US Region                          
Chicken                                            Middle Atlantic     4.336661e+09   
                                                   Mountain            4.336382e+09   
                                                   New England         4.336066e+09   
                                                   South Atlantic      4.336358e+09   
                                                   West North Central  4.336352e+09   
                                                   West South Central  4.336209e+09   
Ham/Pork                                           East North Central  4.336902e+09   
                                                   East South Central  4.336910e+09   
                                                   Middle Atlantic     4.336872e+09   
                                                   Mountain            4.336759e+09   
                                                   Pacific             4.336555e+09   
                                                   South Atlantic      4.336672e+09   
                                                   West North Central  4.336739e+09   
                                                   West South Central  4.337632e+09   
I don't know                                       Pacific             4.336162e+09   
                                                   West North Central  4.336084e+09   
Other (please specify)                             East North Central  4.336889e+09   
                                                   East South Central  4.337105e+09   
                                                   Middle Atlantic     4.336572e+09   
                                                   New England         4.337188e+09   
                                                   Pacific             4.336706e+09   
                                                   South Atlantic      4.336698e+09   
                                                   West North Central  4.336682e+09   
                                                   West South Central  4.336934e+09   
Roast beef                                         East South Central  4.337424e+09   
                                                   Middle Atlantic     4.336073e+09   
                                                   Pacific             4.335958e+09   
                                                   South Atlantic      4.336449e+09   
Tofurkey                                           East North Central  4.337490e+09   
                                                   Middle Atlantic     4.336841e+09   
                                                   Mountain            4.336740e+09   
                                                   New England         4.336895e+09   
                                                   Pacific             4.336958e+09   
                                                   South Atlantic      4.336899e+09   
                                                   West North Central  4.336658e+09   
                                                   West South Central  4.337372e+09   
Turducken                                          Middle Atlantic     4.336739e+09   
                                                   Pacific             4.337475e+09   
Turkey                                             East North Central  4.336802e+09   
                                                   East South Central  4.336830e+09   
                                                   Middle Atlantic     4.336824e+09   
                                                   Mountain            4.336782e+09   
                                                   New England         4.336825e+09   
                                                   Pacific             4.336465e+09   
                                                   South Atlantic    

# Exploring main dishes by locality

In [21]:
pd.crosstab(data['How would you describe where you live?'], 
            data['What is typically the main dish at your Thanksgiving dinner?'])

What is typically the main dish at your Thanksgiving dinner?  Chicken  \
How would you describe where you live?                                  
Rural                                                               2   
Suburban                                                            3   
Urban                                                               7   

What is typically the main dish at your Thanksgiving dinner?  Ham/Pork  \
How would you describe where you live?                                   
Rural                                                                7   
Suburban                                                            17   
Urban                                                                4   

What is typically the main dish at your Thanksgiving dinner?  I don't know  \
How would you describe where you live?                                       
Rural                                                                    3   
Suburban                                                                 1   
Urban                                                                    0   

What is typically the main dish at your Thanksgiving dinner?  Other (please specify)  \
How would you describe where you live?                                                 
Rural                                                                              9   
Suburban                                                                          13   
Urban                                                                             13   

What is typically the main dish at your Thanksgiving dinner?  Roast beef  \
How would you describe where you live?                                     
Rural                                                                  1   
Suburban                                                               3   
Urban                                                                  6   

What is typically the main dish at your Thanksgiving dinner?  Tofurkey  \
How would you describe where you live?                                   
Rural                                                                3   
Suburban                                                             9   
Urban                                                                8   

What is typically the main dish at your Thanksgiving dinner?  Turducken  \
How would you describe where you live?                                    
Rural                                                                 2   
Suburban                                                              1   
Urban                                                                 0   

What is typically the main dish at your Thanksgiving dinner?  Turkey  
How would you describe where you live?                                
Rural                                                            189  
Suburban                                                         449  
Urban                                                            198